project structure

toxdes opened the pull request:
For #1 Created a basic `Hello world` project, with typescript template. Suggestions are welcome.

toxdes committed:
Default react-native typescript template

created with `npx react-native init mobile --template
react-native-template-typescript`

toxdes committed:
Hello, world!

Deleted default template code, and added a bare minimum hello world app.

ankitbko commented on src/mobile/package.json:
```diff
@@ -0,0 +1,44 @@
+    "jest": "^26.6.3",
+    "metro-react-native-babel-preset": "^0.64.0",
+    "react-test-renderer": "17.0.1",
+    "typescript": "^3.8.3"
```
Can we use latest version?

ankitbko commented on src/mobile/app.json:
```diff
@@ -0,0 +1,4 @@
+{
+  "name": "mobile",
```
Do we want to change the name?

ankitbko commented on src/mobile/tsconfig.json:
```diff
@@ -0,0 +1,63 @@
+    // "checkJs": true,                       /* Report errors in .js files. */
+    "jsx": "react-native",                    /* Specify JSX code generation: 'preserve', 'react-native', or 'react'. */
+    // "declaration": true,                   /* Generates corresponding '.d.ts' file. */
+    // "sourceMap": true,                     /* Generates corresponding '.map' file. */
```
enable this

ankitbko commented on src/mobile/tsconfig.json:
```diff
@@ -0,0 +1,63 @@
+
+    /* Strict Type-Checking Options */
+    "strict": true,                           /* Enable all strict type-checking options. */
+    // "noImplicitAny": true,                 /* Raise error on expressions and declarations with an implied 'any' type. */
```
and this

ankitbko commented on src/mobile/tsconfig.json:
```diff
@@ -0,0 +1,63 @@
+    // "noImplicitThis": true,                /* Raise error on 'this' expressions with an implied 'any' type. */
+    // "alwaysStrict": true,                  /* Parse in strict mode and emit "use strict" for each source file. */
+
+    /* Additional Checks */
```
and all these

ankitbko commented on src/mobile/tsconfig.json:
```diff
@@ -0,0 +1,63 @@
+    // "inlineSourceMap": true,               /* Emit a single file with source maps instead of having a separate file. */
+    // "inlineSources": true,                 /* Emit the source alongside the sourcemaps within a single file; requires '--inlineSourceMap' or '--sourceMap' to be set. */
+
+    /* Experimental Options */
```
and these

Cinnybun replied:
Should I click Resolve converstation?

ankitbko commented on src/mobile/.prettierrc.js:
```diff
@@ -0,0 +1,7 @@
+  singleQuote: true,
+  trailingComma: 'all',
+  arrowParens: 'avoid',
+};
```
add following
```
"printWidth": 128,

```

ankitbko commented:
Please add a `setup.md` and document list of softwares (android sdk, android platform version, node ver etc) which are necessary to build this app. And probably also include instructions on how to build/test/run the app.

Cinnybun reviewed (dismissed):
LGTM

toxdes committed:
Made required changes to configs

1. Bumped up typescript to use latest version. (It was lower in default
   template)
2. Changed `printWidth` in `prettierrc`
3. Enabled specified options in `tsconfig.json`.

toxdes committed:
Added `setup.md` and `build:android` command

ankitbko commented on src/mobile/setup.md:
```diff
@@ -0,0 +1,36 @@
+# Development
```
Also add the prerequisites - 
1. Android SDK
2. JAVA

Also add how to generate avd for android emulator after installing SDK -

```
sdkmanager --install 'system-images;android-29;google_apis_playstore;x86_64'
sdkmanager --install 'extras;intel;Hardware_Accelerated_Execution_Manager'

.\avdmanager.bat create avd -f -n crimsonAVD -d 17 -b google_apis/x86_64 --package 'system-images;android-29;google_apis;x86_64'
```

ankitbko replied:
And how to run the emulator - 

```
emulator -avd crimsonAVD
```

toxdes committed:
Added emulator specific instructions in `setup.md`

toxdes committed:
Fixed typos in `setup.md`.

toxdes committed:
Link to fix for metro-bundler connection issue

toxdes committed:
Removed [Dd]ebug from root .gitignore

Since we need to include a dir that has name debug for react-native
project.

toxdes committed:
Local README.md added and modified `tsconfig.json`

Modified `tsconfig.json` to disable `sourceMap` since we are using
`inlineSourceMap` option.

toxdes committed:
Modified READMEs to fix links.

tarunsingh495 commented:
Tested locally. Works fine for me.

ME-ON1 reviewed (approved):
looks fine to me

tarunsingh495 reviewed (approved):
tested Locally . Works fine for me.
